legend-engine-xt-changetoken-compile/src/main/java/org/finos/legend/engine/changetoken/generation/GenerateCastUtil.java

```diff
@@ -0,0 +1,99 @@
+//  Copyright 2022 Goldman Sachs
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package org.finos.legend.engine.changetoken.generation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class GenerateCastUtil
```

Comment:
I think you're going to want better error messages here. For example, on line 36 if key isn't a String, you're going to get a pretty uninformative error message. Ideally, the error messages would give you both the problematic key and the segment of the path that was successfully traversed.